### PR TITLE
#36: Rework class internals

### LINKED ISSUE #36: Rework camera class internals

### LINKED ISSUE #41: FPS camera should move on a plane, even when looking down/upwards

### LINKED ISSUE #45: Make shift key modify speed

### LINKED ISSUE #49: Abstract input processing in main scene view to a new class

### LINKED ISSUE #51: Add static camera

### PR DESCRIPTION
**Pull request by TransientTetra**
Resolves #36; resolves #51; resolves #49; resolves #45; resolves #41

### CODE REVIEW

File: include/view/movement.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#ifndef PROC_GEN_OPENGL_MOVEMENT_HPP
+#define PROC_GEN_OPENGL_MOVEMENT_HPP
+
+
+#include <SDL_events.h>
```
Comment (szymonzygzula) [inline include/view/movement.hpp]: As I wrote to you in dm, not sure why it is not "#include <SDL2/SDL_events.h>"

File: src/view/standard_fps_movement.cpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
+#include "view/standard_fps_movement.hpp"
+#include <SDL.h>
```
Comment (szymonzygzula) [inline src/view/standard_fps_movement.cpp]: As I mentioned before, not sure why it isn't "#include <SDL2/SDL.h>"

**Review by szymonzygzula — CHANGES_REQUESTED** [review]
All clear. Let me know what do you think about those iclude statements and you can merge that

File: include/view/movement.hpp (hunk 1 above)
Comment (TransientTetra) [reply include/view/movement.hpp]: Done

File: src/view/standard_fps_movement.cpp (hunk 2 above)
Comment (TransientTetra) [reply src/view/standard_fps_movement.cpp]: Done

**Review by szymonzygzula — APPROVED** [review]
Great, you can merge

**Pull request merged**